src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UpdateContactDetailsContestedMidHandler.java

```diff
@@ -38,7 +48,46 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
             caseDetails.getId());
 
         FinremCaseData caseData = caseDetails.getData();
+        List<String> errors = new ArrayList<>();
+        errors.addAll(postalService.validate(caseData));
+        errors.addAll(validateCaseData(caseData));
         return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder()
-            .data(caseData).errors(postalService.validate(caseData)).build();
+            .data(caseData).errors(errors).build();
+    }
+
+    private List<String> validateCaseData(FinremCaseData caseData) {
+
+        List<String> errors = new ArrayList<>();
+        ContactDetailsWrapper wrapper = caseData.getContactDetailsWrapper();
+
+        if (caseData.isApplicantRepresentedByASolicitor()) {
+            if (wrapper.getApplicantSolicitorAddress() != null
+                && ObjectUtils.isEmpty(wrapper.getApplicantSolicitorAddress().getPostCode()))  {
```

Comment:
It's not a big problem but it can be made it more readable.

such as 
`Optional.ofNullable(wrapper.getApplicantSolicitorAddress())
            .map(Address::getPostCode)
            .map(String::isEmpty)
            .orElse(false)`

https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=1756&id=finrem-case-orchestration&open=AZB4fiLN2_G31vFDdufu